### PR TITLE
added function subreddit-hot for retrieving hottest posts

### PR DESCRIPTION
**Pull request by jakuzure**

### CODE REVIEW

**Comment by ThatGuyHughesy** [comment]
@jakuzure Cheers for the pull request!

File: src/creddit/client.clj [hunk 1]
```diff
@@ -119,6 +119,12 @@
     (-> (http-get credentials (str "https://www.reddit.com/r/" subreddit "/top/.json?limit=" limit "&t=" (name time)))
         (parse-response))))
 
+(defn subreddit-hot
+  [credentials subreddit limit]
+  (if (valid-limit? limit)
+    (-> (http-get credentials (str "https://www.reddit.com/r/" subreddit "/top/.json?limit=" limit))
```
Comment (ThatGuyHughesy) [inline src/creddit/client.clj]: I assume this should say `/hot` not `/top`?

Comment (jakuzure) [reply src/creddit/client.clj]: Right, overlooked that one!

File: test/creddit/test/client.clj [hunk 2]
```diff
@@ -119,6 +119,16 @@
     (is (thrown? Exception (client/subreddit-top creddit-client "aww" "10" :month)))
     (is (thrown? Exception (client/subreddit-top creddit-client "aww" 10 "month")))))
 
+(deftest test-subreddit-hot
+  (testing "Retrieve hottest subreddit posts"
+    (is (= (with-fake-routes
+             {"https://www.reddit.com/r/aww/top/.json?limit=10"
```
Comment (ThatGuyHughesy) [inline test/creddit/test/client.clj]: This should probably be `/hot` as well

Comment (jakuzure) [reply test/creddit/test/client.clj]: Indeed, I'm having some trouble getting everything to work after changing everything to /hot

Comment (ThatGuyHughesy) [reply test/creddit/test/client.clj]: How are you getting on with this? What exactly is giving you trouble?

Comment (jakuzure) [reply test/creddit/test/client.clj]: The functionality works when changing every occurence of top to hot, but I'm having some trouble getting the test to pass

**Comment by ThatGuyHughesy** [comment]
Hey @jakuzure do you want to push what you have so far and we can try figure out why the tests aren't passing?

**Comment by jakuzure** [comment]
Hi @ThatGuyHughesy, that's what I have so far. The code should be working, but the test isn't quite yet

File: test/creddit/test/client.clj [hunk 3]
```diff
@@ -119,6 +119,16 @@
...
 
+(deftest test-subreddit-hot
+  (testing "Retrieve hottest subreddit posts"
+    (is (= (with-fake-routes
+             {"https://www.reddit.com/r/aww/hot/.json?limit=10"
+              (fn [request]
+                {:status 200 :headers {} :body reddit-response})}
+             (client/subreddit-hot creddit-client "aww" 10))
+           parsed-reddit-response))
+    (is (thrown? Exception (client/subreddit-hot creddit-client "aww" 10)))))
```
Comment (ThatGuyHughesy) [inline test/creddit/test/client.clj]: I think this is the issue, this check is expecting an error to be thrown but the request is valid. Try changing `10` to `"10"` and it should pass as `"10"` is not a valid limit!

Comment (jakuzure) [reply test/creddit/test/client.clj]: You're right of course, passing now! Thanks very much for all the hand-holding :)

Comment (ThatGuyHughesy) [reply test/creddit/test/client.clj]: No problem! Glad you go it working 😄

**Comment by ThatGuyHughesy** [comment]
Can you just squash your commits and then it's ready to merge! The link on how to squash your commits, if you haven't before, is in the CONTRIBUTING.md

**Pull request closed without merging**